src/seedu/addressbook/Main.java

```diff
@@ -104,16 +108,16 @@ private void recordResult(CommandResult result) {
      *
      * @param command user command
      * @return result of the command
+     * @throws StorageOperationException if there are errors writing to the storage file
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageOperationException  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
-            ui.showToUser(e.getMessage());
-            throw new RuntimeException(e);
+        } catch (StorageOperationException e) {
+            throw new StorageOperationException("Error writing to storage file: file may be set to read only.");
```

Comment:
hmm, usually we don't catch and throw the same exception because you wouldn't know if you missed out the implementation and show the stack trace to user. Good attempt otherwise.